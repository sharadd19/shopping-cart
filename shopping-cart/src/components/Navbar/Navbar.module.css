.navbar {
    display: flex;
    justify-content: space-between;
}

.navbarLeft, .navbarRight {
    display: flex;
    align-items: center;
    gap: 100px;
    
}

.title {

}

.store {
    
}