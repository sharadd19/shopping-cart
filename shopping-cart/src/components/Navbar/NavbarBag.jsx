export default function NavbarBag({numberOfItemsInBag}) {
  return (
    <div className="bag">
      <img src="" alt="" className="bag-icon" />
      <p>{/* number of items in bag */}</p>
    </div>
  );
}
