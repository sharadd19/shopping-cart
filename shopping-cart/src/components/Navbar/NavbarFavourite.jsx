/* export default function NavbarFavourite({numberOfFavouriteItems}) {
  return (
    <div className="favourite">
      <img src="" alt="" className="favourite-icon" />
      <p>{number of items in favourite}</p>
    </div>
  );
}
 */