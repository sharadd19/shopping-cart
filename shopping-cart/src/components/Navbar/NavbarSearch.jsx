export default function NavbarSearch() {
    return (
        <div className="search">
        <img src="" alt="" className="icon" />
        <input type="text" placeholder="Search" className="search-query" />
      </div>
    )
}