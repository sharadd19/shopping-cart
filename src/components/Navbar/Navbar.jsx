import { Link } from "react-router-dom";
import styles from "./Navbar.module.css"

export default function Navbar({numberOfBagItems}) {

  return (
    <>
      <div className={styles.navbar}>
        <div className={styles.navbarLeft}>
          <Link to="/">
            <h1 className={styles.title}>SATO.</h1>
          </Link>
          <Link to="/store">
            <h3 className={styles.store}>Store</h3>
          </Link>
        </div>
        <div className={styles.navbarRight}>
          <Link to="/bag">
            <h3 data-testid="number-bag-items">Bag ({numberOfBagItems})</h3>
          </Link>
        </div>
      </div>
    </>
  );
}
