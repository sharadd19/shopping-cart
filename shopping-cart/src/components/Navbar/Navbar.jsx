/* import NavbarFavourite from "./NavbarFavourite";
import NavbarBag from "./NavbarBag"; */

export default function Navbar() {


  /* const numberOfFavouriteItems = 5;
  const numberOfItemsInBag = 5; */
  return (
    <>
      <div className="navbar-left">
        <h1 className="title">fruits.</h1>
        <h3 className="store">Store</h3>
      </div>

     {/*  <div className="navbar-right">
        <NavbarFavourite numberOfFavouriteItems={numberOfFavouriteItems} /> 
        <NavbarBag numberOfItemsInBag={numberOfItemsInBag} />
      </div> */}
    </>
  );
}
