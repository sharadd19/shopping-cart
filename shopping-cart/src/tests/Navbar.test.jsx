import { render, screen } from "@testing-library/react";
import Navbar from "../components/Navbar/Navbar";
import { describe, it, expect } from "vitest";
import { MemoryRouter } from "react-router-dom";

describe("Navbar Component", () => {
    it("navlinks should direct to correct path", () => {
      render(
        <MemoryRouter>
          <Navbar />
        </MemoryRouter>
      );
  
      const logoLink = screen.getByRole("link", { name: /fruits./i });
      expect(logoLink).toHaveAttribute("href", "/");
  
      const storeLink = screen.getByRole("link", { name: /store/i });
      expect(storeLink).toHaveAttribute("href", "/store");
  
      const bagLink = screen.getByRole("link", { name: /bag/i });
      expect(bagLink).toHaveAttribute("href", "/bag");
    });
  
    it("bag items should have correct number of products in them", () => {
      render(
        <MemoryRouter>
          <Navbar numberOfBagItems={4} />
        </MemoryRouter>
      );
  
      expect(screen.getByTestId("number-bag-items").textContent).toBe(
        `Bag (${4})`
      );
    });
  
    it("should render nav and match snapshot", () => {
      const { container } = render(
        <MemoryRouter>
          <Navbar numberOfBagItems={""} />
        </MemoryRouter>
      );
      expect(container).toMatchSnapshot();
    });
  });
  