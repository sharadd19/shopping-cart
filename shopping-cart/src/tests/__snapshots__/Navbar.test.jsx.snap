// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar Component > should render nav and match snapshot 1`] = `
<div>
  <div
    class="_navbar_278c00"
  >
    <div
      class="_navbarLeft_278c00"
    >
      <a
        href="/"
      >
        <h1
          class="_title_278c00"
        >
          fruits.
        </h1>
      </a>
      <a
        href="/store"
      >
        <h3
          class="_store_278c00"
        >
          Store
        </h3>
      </a>
    </div>
    <div
      class="_navbarRight_278c00"
    >
      <a
        href="/bag"
      >
        <h3
          data-testid="number-bag-items"
        >
          Bag (
          )
        </h3>
      </a>
    </div>
  </div>
</div>
`;
